### PR TITLE
Pc/uw schedule api

### PR DESCRIPTION
**Pull request by PeterChu2**
added retrofit uw api for course schedules - note it doesn't return all the schedules for a given term, instead there's an api for all the schedules for a given term AND subject (e.g. ECE)

### CODE REVIEW

File: app/src/main/java/com/example/peterchu/watplanner/MainActivity.java [hunk 1]
```diff
@@ -37,13 +39,32 @@ protected void onCreate(Bundle savedInstanceState) {
...
             @Override
             public void onFailure(Call<CourseResponse>call, Throwable t) {
                 // Log error here since request failed
-                Log.e("MyActivity", t.toString());
+                Log.e("MainActivity", t.toString());
+            }
+        });
+
+        Call<CourseScheduleResponse> scheduleCall = apiService.getSubjectCourseSchedules(
+                "1175", "ECE", Constants.API_KEY
```
Comment (timothytong) [inline app/src/main/java/com/example/peterchu/watplanner/MainActivity.java]: Note: sooner or later we'll have to replace the hardcoded 1175 to a dynamically generated string (which should be easy)

File: app/src/main/java/com/example/peterchu/watplanner/Networking/ApiInterface.java [hunk 2]
```diff
@@ -14,4 +15,20 @@
...
+    @GET("terms/{termId}/{subject}/{catalogNumber}/schedule.json")
+    Call<CourseScheduleResponse> getCourseSchedule(
+            @Path("termId") String termId,
+            @Path("subject") String subject,
+            @Path("catalogNumber") String catalogNumber,
+            @Query("key") String apiKey
+    );
+
+    @GET("terms/{termId}/{subject}/schedule.json")
+    Call<CourseScheduleResponse> getSubjectCourseSchedules(
```
Comment (timothytong) [inline app/src/main/java/com/example/peterchu/watplanner/Networking/ApiInterface.java:28]: so does this API return for example, the schedules of ALL ECE courses?

File: app/src/main/java/com/example/peterchu/watplanner/MainActivity.java (hunk 1 above)
Comment (PeterChu2) [reply app/src/main/java/com/example/peterchu/watplanner/MainActivity.java]: yeah, definitely

File: app/src/main/java/com/example/peterchu/watplanner/Networking/ApiInterface.java (hunk 2 above)
Comment (PeterChu2) [reply app/src/main/java/com/example/peterchu/watplanner/Networking/ApiInterface.java:28]: yup

**Comment by timothytong** [comment]
LGTM! Intern strikes again!

**Pull request merged**